hydra_ros/src/frontend/ros_frontend_publisher.cpp

```diff
@@ -79,9 +90,36 @@ void RosFrontendPublisher::call(uint64_t timestamp_ns,
   if (backend_input.mesh_update) {
     backend_input.mesh_update->timestamp_ns = timestamp_ns;
     mesh_update_pub_->publish(*backend_input.mesh_update);
+    delta_queue_.push(backend_input.mesh_update->sequence_number);
+    stored_delta_.insert(
+        {backend_input.mesh_update->sequence_number, *backend_input.mesh_update});
+    if (config.mesh_delta_queue_size > 0 &&
+        delta_queue_.size() > static_cast<size_t>(config.mesh_delta_queue_size)) {
+      auto first_seq = delta_queue_.front();
+      stored_delta_.erase(first_seq);
+      delta_queue_.pop();
+    }
   }
 
   dsg_sender_->sendGraph(graph, rclcpp::Time(timestamp_ns));
 }
 
+void RosFrontendPublisher::processMeshDeltaQuery(
+    const MeshDeltaSrv::Request::SharedPtr req,
+    MeshDeltaSrv::Response::SharedPtr resp) {
+  LOG(INFO) << "Received request for " << req->sequence_numbers.size()
+            << " mesh deltas...";
+  for (const auto& seq : req->sequence_numbers) {
+    kimera_pgmo_msgs::msg::MeshDelta msg;
```

Comment:
This is probably better as `auto& msg = resp->deltas.emplace_back()` (also doesn't the message needs to be pushed back for this to do something)?